### ISSUE TITLE
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1167: ordinal not in range(128)

### ISSUE THREAD

**Issue by Earnestly** [tehmaze/diagram#14]
* python 3.6.6

When attempting to install this via `python setup.py install --root="$PWD"/t --optimize=1` I get the following traceback:

```
Traceback (most recent call last):
  File "setup.py", line 12, in <module>
    long_description=read('README.md'),
  File "setup.py", line 6, in read
    return open(filename).read()
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1167: ordinal not in range(128)
```

This seems to be related to the `read` function definition in `setup.py`.  Modifying the `open` function to use `encoding='utf-8'` allows the setup to continue without issue.

```diff
diff --git a/setup.py b/setup.py
index 73dbd74..1f3b15b 100644
--- a/setup.py
+++ b/setup.py
@@ -3,7 +3,7 @@ from setuptools import setup
 import os
 
 def read(filename):
-    return open(filename).read()
+    return open(filename, encoding='utf-8').read()
 
 setup(
     name='diagram',
```

I'm not sure if this usage would be portable to python2, so I've left it here as a "works for me" instead.

**Comment by Earnestly** [comment]
A better solution, perhaps, might be to `export PYTHONIOENCODING=utf8`. My build environment intentionally is stripped to the bare minimum, such as `HOME` and `PATH`.  This means I have no `LANG` or `LC_ALL` set (which is a UTF-8 charset by day).  The purpose of this is to try an expose unassumed dependencies which may be why it's tripping up `setup.py` which is falling back to the `C` locale.

I'm not sure what you'd prefer here, either explicitly define the encoding in `setup.py`, document the need for `PYTHONIOENCODING`, or document the requirement on `LANG` or `LC_ALL` to be configured with an available UTF-8 based locale.

**Comment by tehmaze** [comment]
#15 was a duplicate, fixed in b0dbb52ff64dbbe27e521013ab13eee2448068f2
